internal/deploy/yaml.go

```diff
@@ -0,0 +1,213 @@
+package deploy
+
+import (
+	"os"
+	"strings"
+
+	"github.com/shipa-corp/ketch/internal/errors"
+	"sigs.k8s.io/yaml"
+
+	ketchv1 "github.com/shipa-corp/ketch/internal/api/v1beta1"
+)
+
+// Application represents the fields in an application.yaml file that will be
+// transitioned to a ChangeSet.
+type Application struct {
+	Version        *string    `json:"version"`
+	Type           *string    `json:"type"`
+	Name           *string    `json:"name"`
+	Image          *string    `json:"image"`
+	Framework      *string    `json:"framework"`
+	Description    *string    `json:"description"`
+	Environment    *[]string  `json:"environment"`
+	RegistrySecret *string    `json:"registrySecret"`
+	Builder        *string    `json:"builder"`
+	BuildPacks     *[]string  `json:"buildPacks"`
+	Processes      *[]Process `json:"processes"`
+	CName          *CName     `json:"cname"`
+	AppUnit        *int       `json:"appUnit"`
+}
+
+type Process struct {
+	Name  string `json:"name"`  // required
+	Cmd   string `json:"cmd"`   // required
+	Units *int   `json:"units"` // unset? get from AppUnit
+	Ports []Port `json:"ports"` // appDeploymentSpec
+	Hooks Hooks  `json:"hooks"`
+}
+
+type Port struct {
+	Protocol   string `json:"protocol"`
+	Port       int    `json:"port"`
+	TargetPort int    `json:"targetPort"`
+}
+
+type Hooks struct {
+	Restart Restart `json:"restart"`
+}
+
+type Restart struct {
+	Before string `json:"before"`
+	After  string `json:"after"`
+}
+
+type CName struct {
+	DNSName string `json:"dnsName"`
+}
+
+var (
+	defaultVersion  = "v1"
+	defaultAppUnit  = 1
+	typeApplication = "Application"
+	typeJob         = "Job"
+
+	errEnvvarFormat = errors.New("environment variables should be in the format - name=value")
+)
+
+// GetChangeSetFromYaml reads an application.yaml file and returns a ChangeSet
+// from the file's values.
+func (o *Options) GetChangeSetFromYaml(filename string) (*ChangeSet, error) {
+	var application Application
+	b, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	err = yaml.Unmarshal(b, &application)
+	if err != nil {
+		return nil, err
+	}
+
+	var envs []ketchv1.Env
```

Comment:
can we use this function? https://github.com/shipa-corp/ketch/blob/main/internal/utils/envs.go

Reply:
Good call. Implemented.